charts/lnd/templates/cleanup-hook.yaml

```diff
@@ -0,0 +1,23 @@
+# Does not work currently
+# When the hook is disabled, the macaroon and tls configmaps are not deleted
+# When the hook is enabled, they get deleted right before the hook runs
```

Comment:
why do you mean by they get deleted right before the hook runs? who's deleting it?

being deleted is what we want I guess.

Reply:
Consider these two scenarios:

- We don't have this cleanup hook enabled. We delete the lnd release. Now we have two configmaps (tls and macaroon) leftover.

- We have this cleanup hook enabled. We delete the lnd release. Now the hook fails by saying that the tls and macaroon configmaps do not exist. (They're getting deleted auto-magically right before this hook runs, I couldn't figure out by what process/job). Which means the `helm uninstall lnd` command gets stuck and ultimately times out.

Reply:
> We delete the lnd release. Now the hook fails by saying that the tls and macaroon configmaps do not exist.

I'm probably missing something, but this would mean that we don't need the hook?

Reply:
> > We delete the lnd release. Now the hook fails by saying that the tls and macaroon configmaps do not exist.
> 
> I'm probably missing something, but this would mean that we don't need the hook?

That's the strange thing. Like I mentioned in the first scenario, if we don't have the hook, then the tls and macaroon do not get deleted. They are leftover despite all other lnd resources getting deleted.

So it seems like the hook only is responsible for this deletion, but for some reason it fails, and then goes into a crashloopbackoff, whereas you'd expect it to just get completed.